pkg/controller/direct/bigtable/materializedview_controller.go

```diff
@@ -0,0 +1,239 @@
+		return nil, fmt.Errorf("building BigtableMaterializedView client: %w", err)
+	}
+	return gcpClient, err
+}
+
+func (m *modelMaterializedView) AdapterForObject(ctx context.Context, reader client.Reader, u *unstructured.Unstructured) (directbase.Adapter, error) {
+	obj := &krm.BigtableMaterializedView{}
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(u.Object, &obj); err != nil {
+		return nil, fmt.Errorf("error converting to %T: %w", obj, err)
+	}
+
+	id, err := krm.NewMaterializedViewIdentity(ctx, reader, obj)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get bigtable instance admin GCP client. Accepts the non-fully qualified project ID.
+	// E.G. "myproject" instead of "projects/myproject"
+	parentProjectID := id.Parent().Parent.ProjectID
+	instanceAdminClient, err := m.client(ctx, parentProjectID)
+	if err != nil {
+		return nil, fmt.Errorf("error creating instance admin client: %w", err)
+	}
+	return &MaterializedViewAdapter{
+		id:        id,
+		gcpClient: instanceAdminClient,
+		desired:   obj,
+	}, nil
+}
+
+func (m *modelMaterializedView) AdapterForURL(ctx context.Context, url string) (directbase.Adapter, error) {
+	// TODO: Support URLs
+	return nil, nil
+}
+
+type MaterializedViewAdapter struct {
+	id        *krm.MaterializedViewIdentity
+	gcpClient *gcp.InstanceAdminClient
+	desired   *krm.BigtableMaterializedView
+	actual    *bigtablepb.MaterializedView
+}
+
+var _ directbase.Adapter = &MaterializedViewAdapter{}
+
+// Find retrieves the GCP resource.
+// Return true means the object is found. This triggers Adapter `Update` call.
+// Return false means the object is not found. This triggers Adapter `Create` call.
+// Return a non-nil error requeues the requests.
+func (a *MaterializedViewAdapter) Find(ctx context.Context) (bool, error) {
+	log := klog.FromContext(ctx)
+	log.V(2).Info("getting BigtableMaterializedView", "name", a.id)
+
+	materializedViewInfo, err := a.gcpClient.MaterializedViewInfo(ctx, a.id.ParentInstanceIdString(), a.id.ID())
+	if err != nil {
+		if direct.IsNotFound(err) {
+			return false, nil
+		}
+		return false, fmt.Errorf("getting BigtableMaterializedView %q: %w", a.id, err)
+	}
+
+	mapCtx := &direct.MapContext{}
+	if mapCtx.Err() != nil {
+		return false, mapCtx.Err()
+	}
+
+	materializedViewInfo.MaterializedViewID = a.id.ID()
+	a.actual = BigtableMaterializedViewInfo_ToBigtableMaterializedView(materializedViewInfo)
+	return true, nil
+}
+
+// Create creates the resource in GCP based on `spec` and update the Config Connector object `status` based on the GCP response.
+func (a *MaterializedViewAdapter) Create(ctx context.Context, createOp *directbase.CreateOperation) error {
+	log := klog.FromContext(ctx)
+	log.V(2).Info("creating MaterializedView", "name", a.id)
+
+	desired := a.desired.DeepCopy()
+	mapCtx := &direct.MapContext{}
+	if mapCtx.Err() != nil {
+		return mapCtx.Err()
+	}
+	mv := BigtableMaterializedViewSpec_v1alpha1_ToProto(mapCtx, &desired.Spec)
+	mv.Name = a.id.ID()
```

Comment:
Just to double check, does bigtable server accept "name" in the CREATE method? Name is normally required in UPDATE but considered as "output-only" in create.